app/bundles/UserBundle/Model/PasswordStrengthEstimatorModel.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\UserBundle\Model;
+
+use Mautic\UserBundle\Event\PasswordStrengthValidateEvent;
+use Mautic\UserBundle\UserEvents;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use ZxcvbnPhp\Zxcvbn as PasswordStrengthEstimator;
+
+class PasswordStrengthEstimatorModel
+{
+    public const MINIMUM_PASSWORD_STRENGTH_ALLOWED = 3;
+
+    private const DICTIONARY = [
+        'mautic',
+        'user',
+        'lead',
+        'bundle',
+        'campaign',
+        'company',
+    ];
+
+    private PasswordStrengthEstimator $passwordStrengthEstimator;
+
+    public function __construct(private EventDispatcherInterface $dispatcher)
+    {
+        $this->passwordStrengthEstimator = new PasswordStrengthEstimator();
+    }
+
+    /**
+     * @param string[] $dictionary
+     */
+    public function validate(?string $password, int $score = self::MINIMUM_PASSWORD_STRENGTH_ALLOWED, array $dictionary = self::DICTIONARY): bool
+    {
+        $isValid = $score <= $this->passwordStrengthEstimator->passwordStrength($password, $this->sanitizeDictionary($dictionary))['score'];
+
+        $passwordStrengthValidateEvent = new PasswordStrengthValidateEvent($isValid, $password);
+        $this->dispatcher->dispatch($passwordStrengthValidateEvent, UserEvents::USER_PASSWORD_STRENGTH_VALIDATION);
+
+        return $passwordStrengthValidateEvent->isValid;
+    }
+
+    /**
+     * @param string[] $dictionary
+     *
+     * @return string[]
+     */
+    private function sanitizeDictionary(array $dictionary): array
```

Comment:
@diaboloshogunate please provide code examples. I don't understand why is this an issue. You can't tell what array will come to the method so the values are being sanitized. I also didn't write this code, just cherry-picked the commits from our Acquia fork to avoid conflicts with another PR that was doing the same thing. And updated it for M5. Meaning, I don't know the reason for it. Perhaps it covers some edge case. This code was already approved and tested in Acquia repo. I'd like to avoid changing such code if there is not enough reason to do it.

Reply:
It's a minor performance improvement to not have empty and duplicate values in the first place if possible.

```
$sanitized = array_unique(array_filter($dictionary));
if (count($sanitized) != count($dictionary)) {
    trigger_error('There are duplicate or empty values in the dictionary.', E_USER_NOTICE);
}

return $sanitized;
```